File: php/testcode/benchmark_test_00383.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00383(BenchmarkRequest $req): BenchmarkResponse {
    $file = '/var/app/uploads/' . basename($req->param('file'));
    $hash = openssl_digest(file_get_contents($file), 'SHA256');
    return BenchmarkResponse::ok($hash);
}

File: rust/testcode/benchmark_test_00383.rs
enum Route { Home, Profile, Settings, Help }

pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let route_str = req.param("route");
    let route = match route_str.as_str() {
        "home" => Route::Home,
        "profile" => Route::Profile,
        "settings" => Route::Settings,
        "help" => Route::Help,
        _ => return super::shared::BenchmarkResponse::bad_request("Unknown route"),
    };
    let path = match route {
        Route::Home => "/", Route::Profile => "/profile",
        Route::Settings => "/settings", Route::Help => "/help",
    };
    super::shared::BenchmarkResponse::ok(&format!("Location: {}", path))
}
